### ISSUE TITLE
Port not updated in host after changing host

### ISSUE THREAD

**Issue by leesoh** [PowerShellEmpire/Empire#264]
## Empire Version
1.5.0

## OS Information (Linux flavor, Python version)
Linux HOSTNAME 4.5.0-kali1-amd64 #1 SMP Debian 4.5.5-1kali1 (2016-06-06) x86_64 GNU/Linux

## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.
When building a listener, after configuring a port (3389) with a host (192.168.1.1), the host shows as https://192.168.1.1:3389. After updating the host (192.168.1.2), the host just shows as https://192.168.1.2 rather than https://192.168.1.2:3389. Did not confirm whether this affected operations, but running a `Set Port 3389` caused the host to update to https://192.168.1.2:3389.

**Comment by HarmJ0y** [comment]
This is normal/expected behavior. Since Empire is currently just http[s], if the host is set to https://IP with no port specification, port 443 is used, while http://IP implies a default port of 80. Hosts should be specified in IP:Port format when setting.

**Comment by leesoh** [comment]
@HarmJ0y When I do "set Host 192.168.1.10:8080", the host then appears as 192.168.1.10:8080:8080. I think I may not have explained what I meant very well so I'll try again. Starting point is Host set to 192.168.1.10:8080

1. Set Host 192.168.1.15, Host appears as http://192.168.1.15, Port appears as 8080
2. Set Port to 8080, Host appears as http://192.168.1.15:8080
3. Set Port to 8081, Host appears as http://192.168.1.15:8081

Basically it just looks like the Host:Port format is updated after setting a new port, but not after setting a new host.

**Comment by leesoh** [comment]
@HarmJ0y I think this is actually indicative of a bigger problem. When you set the port before the host, the host doesn't include the port (https://<ip> rather than https://<ip>:<port>). If you then execute the listener, you'll see something like this:
```
  ID    Name              Host                                 Type      Delay/Jitter   KillDate    Redirect Target
  --    ----              ----                                 -------   ------------   --------    ---------------
  1     Initial           https://192.168.121.138:8080         native    5/0.0                      
  2     Ini2              https://192.168.121.138:8081         native    5/0.0                      
  3     Ini3              https://192.168.121.138              native    5/0.0                      
```
A netstat will show the above three listening ports, but when you generate a launcher and run it on the host, tcpdump shows traffic bound for the default https port (443) rather than the custom one you set (8082).

TL;DR: If you set the Host after you set a custom Port, your launcher will not use that custom port and will instead default to 443 for HTTPS and, I'm assuming, 80 for http.

**Comment by leesoh** [comment]
In rereading your prior response, I still feel that some backend logic should be added to this to resolve. Either automatically append the port when setting the host, or remove the second Port value altogether and just rely on a Host:Port value. At the very least the Description field should be updated to include this guidance, although I feel the first option would be the most resilient and intuitive.

**Comment by leesoh** [comment]
And one more update, if I first set the `Port` value to 8080, and then set the `Host` value using the <Host>:<IP> format (192.168.121.138:8080), I get a `Host` value of https://192.168.121.138:8080:8080.

**Comment by leesoh** [comment]
This behavior is the same in 2.0 beta.

**Comment by leesoh** [comment]
This has been fixed in #324
